### DISCUSSION TITLE
generation of flow by the turbine in a media at rest

### DISCUSSION THREAD

**Discussion by iliuili** [Exawind/amr-wind#470] [Q&A] [answered]
What is the best testcase for modelling the flow generated by a propeller (in a media initially at rest) with a given rotation rate? 
The problem is quite similar to the wind turbine and in a certain sense "opposite".

**Comment by iliuili** [comment]
As far as I could understand, ICNS.source_terms  must be set properly for turbine simulations.
For the turbine modeled with Actuator Disk  subject to Free Stream the correct options are
incflo.physics = FreeStream Actuator
ICNS.source_terms = ActuatorForcing

The description of ICNS.source_terms can be found at https://amr-wind.readthedocs.io/en/latest/user/inputs.html , but the link for the full list of options produces 404 page :
"Activates source terms for the incompressible Navier-Stokes momentum equations. These strings can be entered in any order with a space between each. Please consult AMR-Wind developer documentation for a comprehensive list of all momentum source terms available."

Could you please give a full list of ICNS.source_terms options , and where the source code of the realization of ActuatorForcing  is located. My first goal is to modify the code to produce the flow past a propeller rotating at a given rate. An after to adjust the model to another geometry of the blades.

Thanks in advance.

**Reply by sayerhs** [reply]
We will need to fix the documentation links. For now you can consult the [restructuredText file](https://github.com/Exawind/amr-wind/blob/main/docs/sphinx/user/inputs_Momentum_Sources.rst) directly for the input parameters. You can also follow this link for [source code](https://github.com/Exawind/amr-wind/tree/main/amr-wind/equation_systems/icns/source_terms) for momentum source terms.

**Comment by sayerhs** [comment, marked as answer]
@iliuili The [momentum sources documentation](https://exawind.github.io/amr-wind/user/inputs_Momentum_Sources.html) page should now be fixed. You can also look at the [Doxygen docs](https://exawind.github.io/amr-wind/api_docs/group__icns__src.html) for a full list of currently implemented momentum sources
